internal/shared/jwt.go

```diff
@@ -61,17 +62,17 @@ func (l LpaStoreClaims) Validate() error {
 		return err
 	}
 
-	if iss == sirius {
-		emailRegex := regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
-		if !emailRegex.MatchString(sub) {
-			return errors.New("Subject is not a valid email")
-		}
-	}
-
-	if iss == mrlpa {
-		uidRegex := regexp.MustCompile("^.+$")
-		if !uidRegex.MatchString(sub) {
-			return errors.New("Subject is not a valid UID")
+	_, isUrn := urn.Parse([]byte(sub))
+
+	if !isUrn {
+		switch iss {
+		case mrlpa:
+			return errors.New("Subject is not a valid URN")
+		case sirius:
+			emailRegex := regexp.MustCompile("^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$")
+			if !emailRegex.MatchString(sub) {
+				return errors.New("Subject is not a valid email or URN")
```

Comment:
Minor comment but in this case the subject is neither a valid email nor URN right? If so, could the error message reflect that?